source/defclass-star.lisp

```diff
@@ -253,29 +265,40 @@ Use the slot name directly:
         (*symbols-to-export* nil)
         (*export-class-name-p* export-class-name)
         (*export-accessor-names-p* export-accessor-names)
-        (*export-slot-names-p* export-slot-names))
+        (*export-slot-names-p* export-slot-names)
+        (*export-predicate-name-p* export-predicate-name)
+        (*predicate-name-transformer* predicate-name-transformer))
     (multiple-value-bind (binding-names binding-values clean-options)
         (extract-options-into-bindings options)
       (progv binding-names (mapcar #'eval binding-values)
         (let ((result (funcall expansion-builder
                                (mapcar 'process-slot-definition slots)
                                clean-options)))
-          (if (or *symbols-to-export*
-                  *export-class-name-p*)
-              `(progn
-                 ,result
-                 (eval-when (:compile-toplevel :load-toplevel :execute)
-                   ;; Don't try to export symbols that don't belong to *package*.
-                   ;; This can happen when inheriting from a class and
-                   ;; overriding some slot.
-                   (export '(,@(remove-if (lambda (sym)
-                                            (not (eq (symbol-package sym) *package*)))
-                                (append (when *export-class-name-p*
-                                          (list name))
-                                 *symbols-to-export*)))
-                           ,(package-name *package*)))
-                 (find-class ',name nil))
-              result))))))
+          `(progn
+             ,@(when *predicate-name-transformer*
+                 (let ((pred-name (funcall *predicate-name-transformer* name)))
+                   `((defun ,pred-name (object)
+                       (subtypep (type-of object) ',name))
```

Comment:
`subtypep` is **much** slower than `typep`. avoid using it wherever `typep` is enough.

Reply:
See my other comment: https://github.com/hu-dwim/hu.dwim.defclass-star/pull/6#discussion_r577418459
`typep` is not enough here, since subtypes must match as well.